src/deployPanel.ts

```diff
@@ -0,0 +1,206 @@
+
+    constructor(
+        extensionPath: string,
+        rpcUri: string,
+        walletExplorer: WalletExplorer,
+        contractDetector: ContractDetector,
+        disposables: vscode.Disposable[],
+        neoExpressConfig?: NeoExpressConfig) {
+
+        this.contractDetector = contractDetector;
+        this.rpcUri = rpcUri;
+        this.walletExplorer = walletExplorer;
+        this.neoExpressConfig = neoExpressConfig;
+        this.viewState = new ViewState();
+
+        this.panel = vscode.window.createWebviewPanel(
+            'deployPanel',
+            (this.neoExpressConfig ? this.neoExpressConfig.basename : this.rpcUri) + ' - Deploy contract',
+            vscode.ViewColumn.Active,
+            { enableScripts: true });
+        this.panel.iconPath = vscode.Uri.file(path.join(extensionPath, 'resources', 'neo.svg'));
+        this.panel.webview.onDidReceiveMessage(this.onMessage, this, disposables);
+
+        const htmlFileContents = fs.readFileSync(
+            path.join(extensionPath, 'src', 'panels', 'deploy.html'), { encoding: 'utf8' });
+        const javascriptHref : string = this.panel.webview.asWebviewUri(
+            vscode.Uri.file(path.join(extensionPath, 'out', 'panels', 'bundles', 'deploy.main.js'))) + '';
+        const cssHref : string = this.panel.webview.asWebviewUri(
+            vscode.Uri.file(path.join(extensionPath, 'out', 'panels', 'deploy.css'))) + '';
+        this.panel.webview.html = htmlFileContents
+            .replace(JavascriptHrefPlaceholder, javascriptHref)
+            .replace(CssHrefPlaceholder, cssHref);
+    }
+
+    public dispose() {
+        this.panel.dispose();
+    }
+
+    private async doDeploy() {
+        try {
+            //
+            // Construct a script that calls Neo.Contract.Create
+            //   -- see: https://docs.neo.org/docs/en-us/reference/scapi/fw/dotnet/neo/Contract/Create.html
+            //
+            // TODO: Find the contracts .abi.json and correctly determine parameters and return types. For now
+            //       we assume that all contracts take two parameters (a string and an array) and return a byte
+            //       array.
+            // TODO: Allow storage usage to be specified in the UI. For now we assume storage is used.
+            // TODO: Allow specification of description, email, etc. in the UI. For now we use blank values.
+            //
+            const sb = neon.default.create.scriptBuilder();
+            const script = sb
+                .emitPush(neon.default.u.str2hexstring('')) // description
+                .emitPush(neon.default.u.str2hexstring('')) // email
+                .emitPush(neon.default.u.str2hexstring('')) // author
+                .emitPush(neon.default.u.str2hexstring('')) // code_version
+                .emitPush(neon.default.u.str2hexstring('')) // name
+                .emitPush(0x01) // storage: {none: 0x00, storage: 0x01, dynamic: 0x02, storage+dynamic:0x03}
```

Comment:
storage and dynamic is part of the aformentioned contract metadata. If the metadata property is missing, prompt the user instead of assuming contracts use storage but arent dynamic